beam-postgres/Database/Beam/Postgres/Syntax.hs

```diff
@@ -1371,6 +1421,23 @@ pgDropExtensionSyntax extName =
 --                                        Nothing
 --      buildJoinFrom tbl newSource Nothing
 
+
+-- TODO
+-- Don't directly expose sum-type constructors?
+-- Constrain 'table' to exist within the query (using s maybe?)
```

Comment:
I think we can probably do some trickery with existential types on `db` to make sure this works. Let me think about it.